materialize-snowflake/staged_file.go

```diff
@@ -133,10 +135,12 @@ func (f *stagedFile) start(ctx context.Context, conn *stdsql.Conn) error {
 	f.group, f.groupCtx = errgroup.WithContext(ctx)
 	f.putFiles = make(chan string)
 
-	// Start the putWorker for this transaction.
-	f.group.Go(func() error {
-		return f.putWorker(f.groupCtx, conn, f.putFiles)
-	})
+	for i := 0; i < MaxConcurrentUploads; i++ {
+		// Start the putWorker for this transaction.
+		f.group.Go(func() error {
+			return f.putWorker(f.groupCtx, conn, f.putFiles)
```

Comment:
I hadn't thought of this before...all these workers are using the same `conn`, which will somewhat diminish their concurrency benefit. They won't actually be able to upload in parallel. They _will_ be able to pick up jobs off the channel though, which will effectively multiply the number of files that can be buffered on disk, so we'd probably never notice a difference. But, it would probably make the most sense for them to get a handle to a `*db` and use that instead of a specific connection.